src/main/java/org/prebid/server/bidder/Bidder.java

```diff
@@ -32,6 +33,19 @@
      */
     Result<List<BidderBid>> makeBids(BidderCall<T> httpCall, BidRequest bidRequest);
 
+    /**
+     * Compound Bidder response with bids and other data to be passed back.
+     * <p>
+     * The errors should contain a list of errors which explain why this bidder's bids will be
+     * "subpar" in some way. For example: the server response didn't have the expected format.
+     */
+    default CompositeBidderResponse makeBidderResponse(BidderCall<T> httpCall, BidRequest bidRequest) {
+        var result = makeBids(httpCall, bidRequest);
+        return result != null
+                ? CompositeBidderResponse.of(result.getValue(), result.getErrors(), null)
+                : null;
+    }
```

Comment:
We don't like this approach. What about to create new interface with method that returns `List<FledgeAuctionConfig>`? Then it can be implemented by adapter class or by other class. (e. g. OpenxBidder implements Bidder<>, InterfaceForFladgeAuctionConfig)

To work with this interface we can add object that implement it to `BidderInstanceDeps`.

Reply:
Thank you, will look into doing this and adding a fledge flag to the yaml config.
Only drawback I see is that it needs to parse the response body twice this way, I would like to avoid that.

Reply:
@CTMBNara, I could use some guidance on the proposed separate interface, as I have encountered a few issues trying to implement it in the way that I understood it.
1) Fledge data is part of the bidder response received with the bids, so it needs to be read when the response body has been deserialized, which only happens inside the adapter - in `makeBids` where they deserialize the response body into their own format. To build `CompositeBidderResponse` from bids and fledge separately outside the bidder, with no state, means the raw response would be parsed twice.
2) `HttpBidderRequester` sees a `Bidder<>` implementation only. To use a new Fledge interface we would have to add a second Fledge parameter to all the methods, OR move the extraction up to `ExchangeService` where the BidderCatalog is available.

I will be on PTO until NY so please feel free to take your time. I will check back from time to time.

Reply:
Discussed with PBS dev team 
We would like to keep your first approach(with default method in Bidder interface)